src/ray/util/util.cc

```diff
@@ -112,7 +120,7 @@ ParseUrlEndpoint(const std::string &endpoint, int default_port) {
     scheme = "tcp://";
   }
   if (scheme == "unix://") {
-#ifdef BOOST_ASIO_HAS_LOCAL_SOCKETS
+#if !defined(_WIN32) && defined(BOOST_ASIO_HAS_LOCAL_SOCKETS)
```

Comment:
This line should be changed to:
```c++
#if BOOST_ASIO_HAS_LOCAL_SOCKETS
```
instead of `#ifdef`.
So you can remove `!defined(_WIN32)` and `defined(xxx)`.

I believe  it's a bug indeed before this PR, but the lower version boost couldn't trigger it.
Image that if `BOOST_ASIO_HAS_LOCAL_SOCKETS` was set to 0, the if block still be executed.